LoopKitUI/Views/DurationPicker.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  DurationPicker.swift
+//  LoopKitUI
+//
+//  Created by Michael Pangburn on 7/15/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+
+public struct DurationPicker: UIViewRepresentable {
+    @Binding public var duration: TimeInterval
+    public var validDurationRange: ClosedRange<TimeInterval>
+    public var minuteInterval: Int
+
+    public init(duration: Binding<TimeInterval>, validDurationRange: ClosedRange<TimeInterval>, minuteInterval: Int = 15) {
+        self._duration = duration
+        self.validDurationRange = validDurationRange
+        self.minuteInterval = minuteInterval
+    }
+
+    public func makeUIView(context: Context) -> UIDatePicker {
+        let picker = UIDatePicker()
+        picker.datePickerMode = .countDownTimer
+        picker.addTarget(context.coordinator, action: #selector(Coordinator.pickerValueChanged(_:)), for: .valueChanged)
+        return picker
+    }
+
+    public func updateUIView(_ picker: UIDatePicker, context: Context) {
+        picker.countDownDuration = duration.clamped(to: validDurationRange)
+        picker.minuteInterval = minuteInterval
+    }
+
+    public func makeCoordinator() -> Coordinator {
+        Coordinator(self)
+    }
+
+    final public class Coordinator {
+        var parent: DurationPicker
```

Comment:
I'm a bit confused about how ownership works here.  Who owns the `DurationPicker`, who owns a `Coordinator`, and how do we make sure there isn't a reference loop here?  (put another way, should this be a `weak` var?)

Reply:
DurationPicker is a struct, so it's a copy rather than a circular reference.
The lifetime of the coordinator is managed by SwiftUI.